dask/dataframe/multi.py

```diff
@@ -919,24 +925,33 @@ def stack_partitions(dfs, divisions, join="outer"):
     dsk = {}
     i = 0
     for df in dfs:
+        df_copy = deepcopy(df)
```

Comment:
Can you say a bit more about why this was needed? The only inplace mutation I see on `dsk`, which we create locally.

Reply:
Without the copy, the `test_append` in `test_multi.py` will fail at the line 
```python
check_with_warning(ddf.a, ddf3.b, df.a, df3.b)
```
Interestingly if the previous 4 checks are removed, the test will not fail. I assumed that this was due to some inplace mutation. Using the copy the test passes. Here is a script for reproduction:

```python
import pandas as pd
import dask.dataframe as dd
import warnings
from dask.dataframe.utils import assert_eq

df = pd.DataFrame({"a": [1, 2, 3, 4, 5, 6], "b": [1, 2, 3, 4, 5, 6]})
df2 = pd.DataFrame(
    {"a": [1, 2, 3, 4, 5, 6], "b": [1, 2, 3, 4, 5, 6]}, index=[6, 7, 8, 9, 10, 11]
)
df3 = pd.DataFrame(
    {"b": [1, 2, 3, 4, 5, 6], "c": [1, 2, 3, 4, 5, 6]}, index=[6, 7, 8, 9, 10, 11]
)

ddf = dd.from_pandas(df, 2)
ddf2 = dd.from_pandas(df2, 2)
ddf3 = dd.from_pandas(df3, 2)

s = pd.Series([7, 8], name=6, index=["a", "b"])

def check_with_warning(dask_obj, dask_append, pandas_obj, pandas_append):
    with warnings.catch_warnings():
        warnings.simplefilter("ignore", FutureWarning)
        expected = pandas_obj.append(pandas_append)

    result = dask_obj.append(dask_append)

    assert_eq(result, expected)

check_with_warning(ddf, s, df, s)
check_with_warning(ddf, ddf2, df, df2)
check_with_warning(ddf.a, ddf2.a, df.a, df2.a)

# # different columns
check_with_warning(ddf, ddf3, df, df3)
check_with_warning(ddf.a, ddf3.b, df.a, df3.b)
```